From bb03aab9ad271c3464f0db158cf8b1fff5b4ddcf Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 19 Jun 2017 11:14:10 +0800
Subject: [PATCH 2686/5242] MLK-16684-1 ARM: dts: enable imx8qm sata

commit  788c7d741ce5865510f6c7b85eec60aba238ef08 from
https://source.codeaurora.org/external/imx/linux-imx.git

- enable imx8qm sata.
- correct sata power supply.
- sata clks:
satahost_clk		hsio_lpcg_sata
phyx1_pclk		phyx1_lpcg
phyx1_epcs_tx_clk	phyx1_lpcg
hyx1_epcs_rx_clk	phyx1_lpcg
phyx2_pclk0		phyx2_lpcg
phyx2_pclk1		phyx2_lpcg

BuildInfo:
- SCFW 9559d5ec, IMX-MKIMAGE 06bc2767, ATF
- U-Boot 2017.03-imx_v2017.03_4.9.51_imx8_beta1+g6dc7b0f

Reviewed-by: Frank Li <frank.li@nxp.com>
Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts |    6 +++
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi    |   58 +++++++++++++++-------
 2 files changed, 47 insertions(+), 17 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
index 3d0909b..3b9e278 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek.dts
@@ -579,3 +579,9 @@
 &isi_3 {
 	status = "okay";
 };
+
+&sata {
+	pinctrl-0 = <&pinctrl_pciea>;
+	clkreq-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 1208a09..85679dd 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -506,26 +506,33 @@
 					reg = <SC_R_PCIE_A>;
 					#power-domain-cells = <0>;
 					power-domains =<&pd_serdes0>;
-				};
-				pd_pcie1: PD_HSIO_PCIE_B {
-					reg = <SC_R_PCIE_B>;
-					#power-domain-cells = <0>;
-					power-domains =<&pd_serdes0>;
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					pd_pcie1: PD_HSIO_PCIE_B {
+						reg = <SC_R_PCIE_B>;
+						#power-domain-cells = <0>;
+						power-domains =<&pd_pcie0>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+					pd_serdes1: PD_HSIO_SERDES_1 {
+						reg = <SC_R_SERDES_1>;
+						#power-domain-cells = <0>;
+						power-domains =<&pd_pcie1>;
+						#address-cells = <1>;
+						#size-cells = <0>;
+
+						pd_sata0: PD_HSIO_SATA_0 {
+							reg = <SC_R_SATA_0>;
+							#power-domain-cells = <0>;
+							power-domains =<&pd_serdes1>;
+						};
+					};
+					};
 				};
 			};
-			pd_serdes1: PD_HSIO_SERDES_1 {
-				reg = <SC_R_SERDES_1>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_hsio>;
-				#address-cells = <1>;
-				#size-cells = <0>;
 
-				pd_sata0: PD_HSIO_SATA_0 {
-					reg = <SC_R_SATA_0>;
-					#power-domain-cells = <0>;
-					power-domains =<&pd_serdes1>;
-				};
-			};
 			pd_gpio: PD_HSIO_GPIO {
 				reg = <SC_R_HSIO_GPIO>;
 				#power-domain-cells = <0>;
@@ -2912,6 +2919,23 @@
 		status = "disabled";
 	};
 
+	sata: sata@5f020000 {
+		compatible = "fsl,imx8qm-ahci";
+		reg = <0x0 0x5f020000 0x0 0x10000>; /* Controller reg */
+		interrupts = <GIC_SPI 88 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&clk IMX8QM_HSIO_SATA_CLK>,
+			 <&clk IMX8QM_HSIO_PHY_X1_PCLK>,
+			 <&clk IMX8QM_HSIO_SATA_EPCS_TX_CLK>,
+			 <&clk IMX8QM_HSIO_SATA_EPCS_RX_CLK>,
+			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_0>,
+			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_1>;
+		clock-names = "sata", "sata_ref", "epcs_tx", "epcs_rx",
+				"phy_pclk0", "phy_pclk1";
+		hsio = <&hsio>;
+		power-domains = <&pd_sata0>;
+		status = "disabled";
+	};
+
 	intmux_cm40: intmux@37400000 {
 		compatible = "nxp,imx-intmux";
 		reg = <0x0 0x37400000 0x0 0x1000>;
-- 
1.7.9.5

